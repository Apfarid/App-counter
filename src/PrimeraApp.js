import React from "react";
import "./index.css";
import PropTypes from "prop-types";

const PrimeraApp = () => {
  const saludo = "Hola";
  return (
    <>
      <h1>{saludo} mundo</h1>
      <p>Mi primera app</p>
    </>
  );
};

PrimeraApp.propTypes = {
  saludo: PropTypes.string,
};

export default PrimeraApp;
