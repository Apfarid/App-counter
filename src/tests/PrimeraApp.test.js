import PrimeraApp from "../PrimeraApp";
// import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

describe("Pruebas en <PrimeraApp/>", () => {
  //   test("debe de mostrar el mensaje Hola, soy Anderson", () => {});
  //   const saludo = "Hola, soy Anderson";
  //   const { getByText } = render(<PrimeraApp saludo={saludo} />);

  //   expect(getByText(saludo)).toBeInTheDocument();

  test("debe de mostrar el mensaje Hola, soy Anderson", () => {
    const saludo = "Hola, soy Anderson";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el subtitilo enviado por props", () => {
    const saludo = "Hola, soy Anderson";
    const subTitulo = " React ";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subTitulo={subTitulo} />
    );

    const textoParrafo = wrapper.find("p").text();
    expect(textoParrafo).toBe(subTitulo);
  });
});
