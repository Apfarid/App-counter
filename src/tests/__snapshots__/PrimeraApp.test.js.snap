// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> debe de mostrar el mensaje Hola, soy Anderson 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Anderson
     
  </h1>
  <p>
     
    React
     
  </p>
</Fragment>
`;
